app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+import datetime
+
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+day_today = datetime.date.today()
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+
+    for friend in friends:
+
+        try:
+            cafe.visit_cafe(friend)
+
+        except VaccineError as v:
+            return str(v)
+
+        except NotWearingMaskError:
+            pass
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+
+        except NotWearingMaskError as n:
+            masks_to_buy = len([friend for friend in friends
+                                if friend.get("wearing_a_mask") is False])
+            return str(n) + f"{masks_to_buy} masks"
```

Comment:
Don't call `visit_cafe` twice. You can catch all exception inside one loop

Reply:
I can, but in this case I will take only first Error which will appearing. In some tests it's important that even if appearing both NotWearingMaskError and VaccineError only VaccineError will raise. Even if NotWearingMaskError appears first. To solve this I made two independent loops. May be you can tell how to make this prioritet ominside one loop?